api/v1alpha1/perses_datasource.go

```diff
@@ -0,0 +1,20 @@
+package v1alpha1
+
+import (
+	"github.com/barkimedes/go-deepcopy"
+	persesv1 "github.com/perses/perses/pkg/model/api/v1"
+)
+
+type Datasource struct {
+	persesv1.Datasource `json:",inline"`
```

Comment:
isn't a bit weird to put the complete datasource object ?

The datasource object also contains the metadata struct and probably each overlap don't you think ? 

How about putting instead `DatasourceSpec` ? That would be the same thing for the dashboard. Using `DashboardSpec` instead of the object `Dashboard` from Perses

Reply:
Yes indeed is more clear that way, I created #10 to fix also the dashboards